### PR TITLE
Refresh the store's models if not found.

### PR DESCRIPTION
**Pull request by howbazaar**
This fix addresses http://pad.lv/1630067. It does that by refreshing the model cache if the store returns a not found error.

A more optimal fix that we should look at later is to add a method on the command base command that does this automatically.

### CODE REVIEW

**Comment by howbazaar** [comment]
## QA Steps

bootstrap lxd controller
start the gui
create a new model
(have to deploy something to get the gui to really create it)
juju models should list it
juju show-model <new name> will now work

similarly, create a model through the gui, and call juju destroy-model <new model>

File: cmd/juju/model/destroy.go [hunk 1]
```diff
@@ -33,6 +36,10 @@ func NewDestroyCommand() cmd.Command {
 // destroyCommand destroys the specified model.
 type destroyCommand struct {
 	modelcmd.ModelCommandBase
+	// RefreshModels is hides the RefreshModels function defined
```
Comment (mjs) [inline cmd/juju/model/destroy.go]: s/ is//

File: cmd/juju/model/destroy.go [hunk 2]
```diff
@@ -17,14 +17,17 @@ import (
...
 	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/jujuclient"
 )
 
 var logger = loggo.GetLogger("juju.cmd.juju.model")
 
 // NewDestroyCommand returns a command used to destroy a model.
 func NewDestroyCommand() cmd.Command {
+	destroyCmd := &destroyCommand{}
+	destroyCmd.RefreshModels = destroyCmd.ModelCommandBase.RefreshModels
```
Comment (mjs) [inline cmd/juju/model/destroy.go]: Can you add a TODO here or somewhere about the more optimal solution?

Comment (howbazaar) [inline cmd/juju/model/destroy.go]: Yes.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**